cli/src/pcluster/resources/head_node/user_data.sh

```diff
@@ -66,7 +66,10 @@ function error_exit
 {
   # wait logs flush before signaling the failure
   sleep 10
-  reason=$(cat /var/log/parallelcluster/chef_error_msg 2>/dev/null) || reason="$1"
+  # trim the error message because there is a size limit of 4096 bytes for cfn-signal
+  # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cloudformation-limits.html
+  cutoff=$(expr 4096 - $(stat --printf="%s" /tmp/wait_condition_handle.txt))
```

Comment:
I recall that the actual limit (at least in the Cfn console), was something less than 1024?

Reply:
This commit is from the PR from @delongmeng. @delongmeng Could you check the actual limits on it?

 I think we also trim the url here in cookbook to make the size:
https://github.com/aws/aws-parallelcluster-cookbook/pull/1688/commits/e87072fe4ee70865b1ca616751c4073558cfd22d

I am going to create a PR to address the security, once the message is static, I think we don't need to worry about this limit.